bundles/core/org.eclipse.smarthome.core.audio/src/main/java/org/eclipse/smarthome/core/audio/internal/AudioServlet.java

```diff
@@ -195,16 +198,50 @@ public URL serve(FixedLengthAudioStream stream, int seconds) {
 
     private URL getURL(String streamId) {
         try {
-            String ipAddress = InetAddress.getLocalHost().getHostAddress(); // we use the primary interface; if a client
-                                                                            // knows it any better, he can himself
-                                                                            // change the url according to his needs.
+            String ipAddress = getLocalNetworkInterface();
+            if (ipAddress == null) {
+                logger.warn("No network interface could be found.");
+                return null;
+            }
             String port = bundleContext.getProperty("org.osgi.service.http.port"); // we do not use SSL as it can cause
                                                                                    // certificate validation issues.
             return new URL("http://" + ipAddress + ":" + port + SERVLET_NAME + "/" + streamId);
-        } catch (UnknownHostException | MalformedURLException e) {
+        } catch (MalformedURLException e) {
             logger.error("Failed to construct audio stream URL: {}", e.getMessage(), e);
             return null;
         }
     }
 
+    /**
+     * Finds the (non loopback, non localhost) local network interface.
+     */
+    public String getLocalNetworkInterface() {
+        try {
+            String localInterface = null;
+            Enumeration<NetworkInterface> interfaces = NetworkInterface.getNetworkInterfaces();
+            while (interfaces.hasMoreElements()) {
+                NetworkInterface current = interfaces.nextElement();
+                if (!current.isUp() || current.isLoopback() || current.isVirtual()) {
```

Comment:
r u sure it's always a good idea to ignore virtual interfaces?
also, as I mentioned in #2331, IMHO it would be good to have a configuration option to override this automatism, as with multiple interfaces such algorithms tend to always pick the wrong one